### PR TITLE
Update changelog

### PR DESCRIPTION
**Pull request by AlyssaCote**
Updated the changelog for release.

### CODE REVIEW

File: doc/changelog.rst [hunk 1]
```diff
@@ -1,28 +1,12 @@
...
 
-Released on *TBD*, 2023
+Released on 14 December, 2023
 
 Description
 
-- Initial release of SmartDashboard
\ No newline at end of file
+- The initial release of SmartDashboard includes capabilities for viewing 
+experiment entity properties and statuses.
```
Comment (MattToast) [inline doc/changelog.rst:12]: Small pedantic suggestion for future readability
```suggestion
- The initial release of SmartDashboard includes capabilities for viewing 
  experiment entity properties and statuses.
```

**Review by MattToast — APPROVED** [review]
One small pedantic change request for potential future readability, but otherwise nothing worth holding approval up over!

LGTM!!

File: doc/changelog.rst (hunk 1 above)
Comment (AlyssaCote) [reply doc/changelog.rst]: fixed!

**Comment by MattToast** [comment]
I'd match the SS change log (iirc 14 December, 2023)

**Pull request merged**